File: native/ios/FireballNative/Core/FireballAnalytics.swift
import Foundation
#if canImport(os)
import os.log
#endif

enum FireballAnalytics {
    static func log(_ event: String, settings: FireballSettings, properties: [String: String] = [:]) {
        guard settings.analytics || settings.loggingEnabled else { return }
        let props = properties.isEmpty ? "" : " \(properties)"
        #if canImport(os)
        Logger(subsystem: "com.fireball.nativeapp", category: "analytics")
            .info("\(event, privacy: .public)\(props, privacy: .public)")
        #else
        print("[FireballAnalytics] \(event)\(props)")
        #endif
    }
}

File: native/android/core-data/src/main/kotlin/com/fireball/nativeapp/core/data/FireballAnalytics.kt
package com.fireball.nativeapp.core.data

import android.util.Log
import com.fireball.nativeapp.core.model.FireballSettings

/** Lightweight analytics gated by settings (no third-party SDK). */
object FireballAnalytics {
    private const val TAG = "FireballAnalytics"

    fun log(event: String, settings: FireballSettings, properties: Map<String, String> = emptyMap()) {
        if (!settings.analytics && !settings.loggingEnabled) return
        val props = if (properties.isEmpty()) "" else " $properties"
        Log.i(TAG, "$event$props")
    }
}
